### ISSUE TITLE
2.18.4 regression: anonymous class properties being inlined with anonymous class declarations

### ISSUE THREAD

**Issue by Ocramius** [FriendsOfPHP/PHP-CS-Fixer#5575]
While upgrading from `2.18.3` to `2.18.4`:

```diff
         $url = $this->createProvider(
             new DummyPayoneGateway(null, function () {
-                throw new class() extends PaymentException {
-                    protected $message = 'my error';
+                throw new class() extends PaymentException { protected $message = 'my error';
                 };
             })
         )->buildPaymentRedirectUrl($order);
```

Overall, that `protected $message` should stay on a new line, I suppose?

`.php_cs.dist`:

```php
<?php

$finder = PhpCsFixer\Finder::create()
    ->in([__DIR__ . '/src', __DIR__ . '/tests']);

return PhpCsFixer\Config::create()
    ->setRules([
        '@Symfony' => true,
        'strict_param' => true,
        'array_syntax' => ['syntax' => 'short'],
        'phpdoc_to_comment' => false,
        'no_unneeded_final_method' => false,
        'no_superfluous_phpdoc_tags' => ['allow_mixed' => true],
    ])
    ->setFinder(
        PhpCsFixer\Finder::create()
            ->in([__DIR__ . '/src', __DIR__ . '/tests'])
    );
```

**Comment by huye** [comment]
2.18.3:

> class className {
> }

2.18.4:
> class className
> {
> }

I want to keep brace on the same line as 2.18.3. How do I set them?

**Comment by kubawerlos** [comment]
@huye why did you post your message here? This issue is about anonymous classes where open brace is not a problem.

**Comment by kubawerlos** [comment]
@Ocramius in `2.18.3` `SingleLineThrowFixer` had priority [`1`](https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/v2.18.3/src/Fixer/FunctionNotation/SingleLineThrowFixer.php#L71) and `BracesFixer` had [`-25`](https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/v2.18.3/src/Fixer/Basic/BracesFixer.php#L133), so `BracesFixer` was adding new lines after `SingleLineThrowFixer` removed them. In `2.18.4` priority of `BracesFixer` got changed to [`35`](https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/v2.18.4/src/Fixer/Basic/BracesFixer.php#L133) so it runs first now.

Take a look at https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/5593 - is that making sense or should it be change the other way - that `SingleLineThrowFixer` is not touching anonymous class definition?

**Comment by Ocramius** [comment]
Totally missed to provide feedback on this, @kubawerlos @keradus!

I'll see if I can test this on a new release, but I'm really unable to provide active help on understanding the underlying cause of the original issue due to lack of knowledge of the codebase :S

**Comment by Ocramius** [comment]
Seems like this is still reproducible in 2.18.6:

```diff
--- Original
+++ New
@@ -229,8 +229,7 @@
 
         $url = $this->createProvider(
             new DummyPayoneGateway(null, function () {
-                throw new class() extends PaymentException {
-                    protected $message = 'my error';
+                throw new class() extends PaymentException { protected $message = 'my error';
                 };
             })
         )->buildPaymentRedirectUrl($order);
      ----------- end diff -----------
```

Should I open a new issue about this one? :thinking:

**Comment by keradus** [comment]
The fix prepared by @kubawerlos  - https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/5593 - was merged into 2.18.7, according to the milestone.

GitHub clsoes the ticket automatically when PR is merged.
